packages/java/parser-jvm-plugin-nonnull/src/main/java/com/vaadin/fusion/parser/plugins/nonnull/NonnullPluginConfig.java

```diff
@@ -0,0 +1,56 @@
+package com.vaadin.fusion.parser.plugins.nonnull;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.vaadin.fusion.parser.core.PluginConfiguration;
+import com.vaadin.fusion.parser.utils.ConfigList;
+
+public class NonnullPluginConfig
+        implements ConfigList<String>, PluginConfiguration {
+    private final Set<String> disable = new HashSet<>();
+    private final boolean disableAllDefaults = false;
+    private final Set<String> use = new HashSet<>();
+
+    public NonnullPluginConfig() {
+    }
+
+    public NonnullPluginConfig(Collection<String> use,
+            Collection<String> disable) {
+        if (disable != null) {
+            this.disable.addAll(disable);
+        }
+
+        if (use != null) {
+            this.use.addAll(use);
+        }
+    }
+
+    @Override
+    public Collection<String> getDisabledOptions() {
+        return disable;
+    }
+
+    @Override
+    public Collection<String> getUsedOptions() {
+        return use;
+    }
+
+    @Override
+    public boolean shouldAllDefaultsBeDisabled() {
+        return disableAllDefaults;
+    }
+
+    static class Processor extends ConfigList.Processor<String> {
+        private static final Set<String> defaults = Set.of(
```

Comment:
My idea here was to allow a specific set of most known `Nonnull` annotation, not all the annotation that has `Nonnull` part in the name. Mostly because we also need to have JetBrains' `NotNull` annotation here.

If the user wants to use any other annotation, they are free to add them to the plugin's configuration.